tests/test_utils.py

```diff
@@ -136,6 +136,64 @@ def test_verify_project_access_ok(client: flask.testing.FlaskClient) -> None:
     utils.verify_project_access(project=project)
 
 
+def test_verify_project_user_key_denied(client: flask.testing.FlaskClient) -> None:
+    """A user must have an entry in projectUserKeys to access a project."""
+    # User
+    user = models.UnitUser.query.filter_by(unit_id=1).first()
+    assert user
+
+    # Get project
+    project = models.Project.query.filter_by(unit_id=1).first()
+    assert project
+
+    # get projectUserKey
+    projectUserKey = models.ProjectUserKeys.query.filter_by(
+        project_id=project.id, user_id=user.username
+    ).one_or_none()
+    assert projectUserKey
+
+    # delete projectUserKey
+    db.session.delete(projectUserKey)
+    db.session.commit()
+
+    # verify no projectUserKey
+    projectUserKey = models.ProjectUserKeys.query.filter_by(
+        project_id=project.id, user_id=user.username
+    ).one_or_none()
+    assert not projectUserKey
+
+    # Set auth.current_user
+    flask.g.flask_httpauth_user = user
+
+    # Verify project access -- not ok
+    with pytest.raises(AccessDeniedError) as err:
+        utils.verify_project_user_key(project=project)
+    assert "You have lost access to this project" in str(err.value)
+
+
+def test_verify_project_user_key_ok(client: flask.testing.FlaskClient) -> None:
+    """A user must have an entry in projectUserKeys to access a project."""
+    # user
+    user = models.UnitUser.query.filter_by(unit_id=1).first()
+    assert user
+
+    # Get project
+    project = models.Project.query.filter_by(unit_id=1).first()
+    assert project
+
+    # get projectUserKey
+    projectUserKey = models.ProjectUserKeys.query.filter_by(
+        project_id=project.id, user_id=user.username
+    ).one_or_none()
+    assert projectUserKey
+
+    # Set auth.current_user
+    flask.g.flask_httpauth_user = user
+
+    # Verify project access -- not ok
```

Comment:
Aside from changing this comment, can you add a check that the error message is not logged? 
```suggestion
    # Verify project access -- ok
```

Reply:
Mmm, But the error is an exception, by not getting raised it should be checked. It is the same as

**test_verify_cli_version_ok** or **test_verify_project_access_ok**

Albeit, it is anyway a 400 error, so no logs should happen in backend anyway. The no ok checks the exception

Reply:
Yep I understand now, I looked at it in the wrong way yesterdat